File: src/Stocks.java
public class Stocks {
    int quantity;

    public synchronized void addQty(int qty){


        System.out.println("Before adding quantity was = "+this.quantity+
            "\nyou added "+qty);
        this.quantity +=qty;
        System.out.println("after adding quantity is now "+this.quantity);
        notifyAll();

    }

    public synchronized void remQty(int qty) throws InterruptedException{


        if(this.quantity > qty){
            System.out.println("Before removing quantity was = "+this.quantity+
            "\nyou removed "+qty);
            this.quantity-=qty;
            System.out.println("after removing quantity is now "+this.quantity);

            notifyAll();
        }


        else{ 
            System.out.println("Quantity is too low");
            wait();
            }

    }

    
}

File: pythonVersion/src/Stocks.py
import threading

class Stocks():
    def __init__(self):
        self.quantity = 0
        self.lock = threading.RLock()
        
    
    def addQty(self, qty):
        with self.lock:
            print(f"before adding, quantity was = {self.quantity}")
            self.quantity += qty
            print(f"you added {qty}")
            print(f"after adding it's now = {self.quantity}")
        
    
    def remQty(self, qty):
        with self.lock:
            if(self.quantity > qty):
                print(f"before removing, quantity was = {self.quantity}")
                self.quantity -= qty
                print(f"you removed {qty}")
                print(f"after removing, quantity is now = {self.quantity}")

                
            else:
                print("impossible to remove")  